src/wallet.c

```diff
@@ -208,6 +211,10 @@ int wallet_generate_key(HDNode *node, const char *keypath, int keypath_len,
     memcpy(kp, keypath, keypath_len);
     kp[keypath_len] = '\0';
 
+    if (kp[0] != 'm' || kp[1] != '/') {
```

Comment:
Do we need a check before that to ensure `strlen(kp) > 1` (or even > 2 with an extra `goto err` in case its not >2)?